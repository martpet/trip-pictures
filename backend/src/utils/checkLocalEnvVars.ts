import { EnvName } from '~/types';

export const checkLocalEnvVars = (envName: EnvName) => {
  const names = [
    'VITE_PERSONAL_SUBDOMAIN',
    'PERSONAL_DEV_ACCOUNT_ID',
    'PERSONAL_DEV_ACCOUNT_REGION',
  ];
  if (envName === 'personal') {
    names.forEach((varName) => {
      if (!process.env[varName]) {
        throw new Error(`environment variable "${varName}" is missing from .env.local`);
      }
    });
  }
};
