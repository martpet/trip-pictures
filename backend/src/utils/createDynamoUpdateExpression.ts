export const createDynamoUpdateExpression = (
  data: Record<string, unknown>,
  parentKey?: string
) => {
  const ExpressionAttributeNames: Record<string, string> = {};
  const ExpressionAttributeValues: Record<string, unknown> = {};
  const actions: string[] = [];

  Object.entries(data).forEach(([attr, val]) => {
    actions.push(`${parentKey ? `#${parentKey}.` : ''}#${attr} = :${attr}`);
    ExpressionAttributeNames[`#${attr}`] = attr;
    ExpressionAttributeValues[`:${attr}`] = val;
    if (parentKey) {
      ExpressionAttributeNames[`#${parentKey}`] = parentKey;
    }
  });

  return {
    UpdateExpression: `set ${actions.join(',')}`,
    ExpressionAttributeNames,
    ExpressionAttributeValues,
  };
};
