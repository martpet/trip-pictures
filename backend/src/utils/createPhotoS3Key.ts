type Props = {
  userId: string;
  fingerprint: string;
};

export function createPhotoS3Key({ userId, fingerprint }: Props) {
  return `${userId}/${fingerprint}`;
}
