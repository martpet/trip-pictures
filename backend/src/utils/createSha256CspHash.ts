import crypto from 'crypto';

export const createSha256CspHash = (content: string) =>
  `sha256-${crypto.createHash('sha256').update(content).digest('base64')}`;
