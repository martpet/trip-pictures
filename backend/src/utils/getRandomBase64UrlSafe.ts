import crypto from 'crypto';
import { promisify } from 'util';

export const getRandomBase64UrlSafe = async (size: number) => {
  const cryptoRandomBytes = promisify(crypto.randomBytes);
  const bytes = await cryptoRandomBytes(size);
  return bytes.toString('base64url');
};
