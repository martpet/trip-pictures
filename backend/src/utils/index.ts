export * from '~/../../shared/utils';
//
export * from './checkLocalEnvVars';
export * from './createDynamoUpdateExpression';
export * from './createPhotoS3Key';
export * from './createSha256CspHash';
export * from './getAllowedOrigins';
export * from './getRandomBase64UrlSafe';
export * from './getRoleCredentials';
export * from './objectValuesToJson';
