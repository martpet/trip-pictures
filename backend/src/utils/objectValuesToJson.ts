import { JsonObject } from 'type-fest';

export const objectValuesToJson = (input: JsonObject) =>
  Object.fromEntries(
    Object.entries(input).map(([key, value]) => [key, JSON.stringify(value)])
  );
