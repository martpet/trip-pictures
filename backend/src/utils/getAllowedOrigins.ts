import { appEnvs, localhostUrl } from '~/consts';
import { getCdkEnv } from '~/utils';

export const getAllowedOrigins = () => {
  const envName = getCdkEnv();
  const { appDomain } = appEnvs[envName];

  const origins = [`https://${appDomain}`];

  if (envName === 'personal') {
    origins.push(localhostUrl);
  }

  return origins;
};
