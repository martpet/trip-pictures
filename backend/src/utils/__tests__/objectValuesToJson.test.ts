import { itReturns } from '~/constructs/Api/utils';

import { objectValuesToJson } from '../objectValuesToJson';

const args = [
  {
    string: '1',
    number: 1,
    array: ['a'],
    object: { foo: 'bar' },
  },
] as Parameters<typeof objectValuesToJson>;

describe('objectValuesToJson', () => {
  itReturns(objectValuesToJson, args);
});
