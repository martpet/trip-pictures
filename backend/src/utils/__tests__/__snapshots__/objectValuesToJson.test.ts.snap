// Vitest Snapshot v1

exports[`objectValuesToJson > returns a correct value 1`] = `
{
  "array": "[\\"a\\"]",
  "number": "1",
  "object": "{\\"foo\\":\\"bar\\"}",
  "string": "\\"1\\"",
}
`;
